Cython/Compiler/TypeInference.py

```diff
@@ -359,9 +359,12 @@ class SimpleAssignmentTypeInferer(object):
     applies to nested scopes in top-down order.
     """
     def set_entry_type(self, entry, entry_type):
-        entry.type = entry_type
-        for e in entry.all_entries():
+        for e in [entry] + entry.all_entries():
```

Comment:
`entry.all_entries()` already includes `entry`, so the code was already redundant before.